Source/Mock/lhc_mock.cpp

```diff
@@ -108,3 +124,38 @@ bool Mock_Internal_HCHttpCallPerformAsync(
 
     return true;
 }
+
+HRESULT Mock_Internal_ReadRequestBodyIntoMemory(
+    _In_ HCCallHandle originalCall,
+    _Out_ http_internal_vector<uint8_t>* bodyBytes
+    )
+{
+    HCHttpCallRequestBodyReadFunction readFunction;
+    size_t bodySize;
+    void* context;
+    RETURN_IF_FAILED(
+        HCHttpCallRequestGetRequestBodyReadFunction(originalCall, &readFunction, &bodySize, &context)
+    );
+
+    http_internal_vector<uint8_t> tempBodyBytes(bodySize);
```

Comment:
is this code safe to throw exceptions in?

Reply:
err, i'm not sure. are you asking because this might throw while allocating?

Reply:
the previous code used an `http_memory_buffer`, happy to change to doing that if it'll avoid this question lol

Reply:
no that doesn't help, let's assume the old code was doing the right thing